test/scripts/agent-images/create_agent_images.sh

```
@@ -19,40 +21,73 @@ if [ -n "${FLIGHTCTL_RPM:-}" ]; then
     BUILD_ARGS="${BUILD_ARGS} --build-arg=RPM_PACKAGE=${RPM_PACKAGE}"
 fi
 
+build_images() {
+    for img in $IMAGE_LIST; do
+        containerfile_path="${SCRIPT_DIR}/Containerfile-e2e-${img}.local"
+        container_name="localhost:5000/flightctl-device:${img}"
 
+        if [ "$BUILD_TYPE" = "regular" ]; then
+            # create a temporary directory and cleanup on exit
+            tmpdir=$(mktemp -d)
+            trap "rm -rf $tmpdir" EXIT
```

Shellcheck (0.10.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)